dev-client/src/hooks/permissionHooks.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {Project, ProjectRole} from 'terraso-client-shared/project/projectSlice';
+
+import {useSelector} from 'terraso-mobile-client/store';
+
+type ProjectFilter = (project: Project) => boolean;
+
+export const useProjectUserRolesFilter = (
+  userRoles?: ProjectRole[],
+): ProjectFilter => {
+  const currentUser = useSelector(state => state.account.currentUser.data);
+  if (userRoles) {
+    return project =>
+      !!Object.values(project.memberships).find(
+        membership =>
+          membership.userId === currentUser?.id &&
+          userRoles.indexOf(membership.userRole) >= 0,
+      );
+  } else {
+    return _ => true;
```

Comment:
I decided that this was friendlier to the usage code rather than finding a way to conditionally invoke the hook, which is a little cumbersome in React since it wants hook invocations to be consistent between renders.